tests/snappi/pfc/files/helper.py

```diff
@@ -519,9 +531,16 @@ def __verify_results(rows,
                                   'Total TX dropped packets {} should be 0'.
                                   format(dropped_packets))
 
-        # Check if pause frames are counted in the PFC counters
-        for peer_port, prios in flow_port_config[1].items():
-            for prio in prios:
-                pfc_pause_rx_frames = get_pfc_frame_count(duthost, peer_port, prio)
+    # Check if pause frames are correctly counted in the PFC counters
+    for peer_port, prios in flow_port_config[1].items():
+        for prio in range(len(prios)):
+            pfc_pause_rx_frames = get_pfc_frame_count(duthost, peer_port, prios[prio])
+            total_egress_packets, _ = get_queue_count(duthost, peer_port, prios[prio])
+            if test_flow_pause and class_enable_vec == class_enable_vector.BIT_SET:
                 pytest_assert(pfc_pause_rx_frames > 0,
                               "PFC pause frames with zero source MAC are not counted in the PFC counters")
+            elif class_enable_vec == class_enable_vector.NO_BIT_SET:
+                pytest_assert(pfc_pause_rx_frames == 0,
+                              "PFC pause frames with no bit set in the class enable vector should be dropped")
+                pytest_assert(total_egress_packets == test_tx_frames[prio], "Queue counters should increment for \
```

Comment:
Why we need this? The comment in L534 says this for loop is used for checking PFC counters.

Reply:
I will make it clearer but we also need to check that regular packets are able to flow through and show up in the egress queue when the class enable vector is not set since the pfc pause frames received by the switch from the ixia is invalid, and the switch will not pause.

Reply:
You can make the information of pytest_assert clearer, e.g., "PFC pause frames with no bit set in the class enable vector should not pause any data packets. And queue counters should increment correspondingly."